### PR TITLE
Fixes #30654 - update batch logic for SP sync

### PR DESCRIPTION
**Pull request by jlsherrill**
this updates the batch logic for smart proxy
syncs in two ways.  It adjusts the batch size default
to 100, as some users were not noticing issues previously
with 1000 repos being synced at a time.  It also sorts the repos
by rpm count, to hopefully put the larger ones at the end

### CODE REVIEW

**Comment by chris1984** [comment]
@jlsherrill is there a good way to test this?

File: app/models/setting/content.rb [hunk 1]
```diff
@@ -92,7 +92,7 @@ def self.default_settings
       self.set('check_services_before_actions', N_("Whether or not to check the status of backend services such as pulp and candlepin prior to performing some actions."),
                true, N_('Check services before actions')),
       self.set('foreman_proxy_content_batch_size', N_("How many repositories should be synced concurrently on the capsule.  A smaller number may lead to longer sync times.  A larger number will increase dynflow load."),
-               25, N_('Batch size to sync repositories in.')),
+               100, N_('Batch size to sync repositories in.')),
```
Comment (johnpmitsch) [inline app/models/setting/content.rb:95]: I guess this setting wasn't being used for anything? :thinking:

Comment (johnpmitsch) [reply app/models/setting/content.rb:95]: oh nvm, ignore me! I see it was moved

**Comment by jlsherrill** [comment]
Testing a capsule sync with multiple repositories and verifying that the larger repos (by rpm count) are synced last  would probably be sufficient 

To fully test, you'd want to test a sync with ~500-1000 small-large repos and cause updates in ~5% of them and compare sync times before and after this change.  I'm not really sure its fully worth that though :)

**Comment by johnpmitsch** [comment]
@jlsherrill getting an error on smart proxy sync:

```
2020-08-20T15:16:59 [E|app|] NoMethodError: undefined method `by_rpm_count' for []:Array
 | /home/vagrant/katello/app/lib/actions/katello/capsule_content/sync_capsule.rb:16:in `block in plan'
 | /home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.6/lib/dynflow/execution_plan.rb:384:in `switch_flow'
 | /home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.6/lib/dynflow/action.rb:425:in `sequence'
 | /home/vagrant/katello/app/lib/actions/katello/capsule_content/sync_capsule.rb:14:in `plan'
 | /home/vagrant/foreman/.vendor/ruby/2.5.0/gems/dynflow-1.4.6/lib/dynflow/action.rb:517:in `block (3 levels) in execute_plan'
```

Seems like it expected AR association instead of Array

**Comment by jlsherrill** [comment]
Note, i haven't tested this fix yet, hold off until i give the :green_circle:

File: app/services/katello/smart_proxy_helper.rb [hunk 2]
```diff
@@ -13,34 +13,32 @@ def default_capsule?
...
         fail _("Lifecycle environment '%{environment}' is not attached to this capsule.") % { :environment => environment.name }
       end
-
-      ret
     end
 
-    private
+    def combined_repos_available_to_capsule(environment = nil, content_view = nil, repository = nil)
+      lifecycle_environment_check(environment, repository
+      )
```
Comment (johnpmitsch) [inline app/services/katello/smart_proxy_helper.rb:26]: Was having this closing ) on a separate line intentional?

Comment (jlsherrill) [reply app/services/katello/smart_proxy_helper.rb:26]: nope!

**Pull request merged**